hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieBaseRelation.scala

```diff
@@ -340,6 +340,21 @@ abstract class HoodieBaseRelation(val sqlContext: SQLContext,
     latestBaseFiles.groupBy(getPartitionPath)
   }
 
+  protected def getSelectedPartitionPaths(
+      globbedPaths: Seq[Path],
+      partitionFilters: Seq[Expression],
+      dataFilters: Seq[Expression]): Seq[Path] = {
+    val partitionDirs = if (globbedPaths.isEmpty) {
+      fileIndex.listFiles(partitionFilters, dataFilters)
+    } else {
+      val inMemoryFileIndex = HoodieInMemoryFileIndex.create(sparkSession, globbedPaths)
+      inMemoryFileIndex.listFiles(partitionFilters, dataFilters)
+    }
+
+    val fsView = new HoodieTableFileSystemView(metaClient, timeline, partitionDirs.flatMap(_.files).toArray)
```

Comment:
Sure, will do

Reply:
Here I move the codes to its caller `collectFileSplits` to reuse `HoodieTableFileSystemView` and `HoodieInMemoryFileIndex`. Also I think this can address the TODO that avoid iterating over listed files multiple times.